mimic/model/nova_objects.py

```diff
@@ -798,6 +798,21 @@ def request_delete(self, http_delete_request, server_id):
         server.update_status(u"DELETED")
         return b''
 
+    def action(self, http_action_request, server_id):
+        """
+        Perform the requested action on the provided server
+        """
+        action_json = loads(http_action_request.content.read())
+        server = self.server_by_id(server_id)
+        if server is None:
+            http_action_request.setResponseCode(500)
+            return b''
+        if 'resize' in action_json:
+            flavor = action_json['resize']['flavorRef']
+            server.flavor_ref = flavor
+            http_action_request.setResponseCode(202)
+            return b''
+
```

Comment:
Only General Purpose and Standard flavor servers can be resized and they can only be resized to a larger flavor.  The API returns a 400 when a request is made to resize a server that is not General Purpose or Standard.  The API returns a 400 when a request is made to resize a server to another flavor class (attempting to resize from a General Purpose flavor class to a Standard flavor class)